### PR TITLE
BUG: Fix length 1 .loc slicing with datetime string GH16071

### LINKED ISSUE #16071: .loc slice with datetime string with DataFrame of length 1 gives an empty DataFrame
I've already raised the problem at https://github.com/pandas-dev/pandas/issues/13929#issuecomment-295326414 but it turned out to be a different issue.

#### The following test code illustrates the problem

```python
import pandas as pd
from pandas.util import testing as tm

from datetime import datetime
df = pd.DataFrame(columns=['1'],
                  index=pd.date_range('2016-10-01T00:00:00',
                                      '2016-10-01T23:59:59'))
df_sliced_dt = df.loc[datetime(2016,10,1):]
tm.assert_frame_equal(df, df_sliced_dt)

df = pd.DataFrame(columns=['1'],
                  index=pd.date_range('2016-10-01T00:00:00',
                                      '2016-10-01T23:59:59'))
df_sliced_str = df.loc['2016-10-01T00:00:00':]
tm.assert_frame_equal(df, df_sliced_str)
```

### PR DESCRIPTION
**Pull request by cpaulik**
I've written the test in pytest style since I read that you are moving to this style. If I misunderstood I can add the test to the class. 

- [x] closes #16071 
 - [x] tests added / passed
 - [x] passes ``git diff upstream/master --name-only -- '*.py' | flake8 --diff``
 - [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/indexing/test_loc.py [hunk 1]
```diff
@@ -628,3 +628,22 @@ def test_loc_empty_list_indexer_is_ok(self):
...
+    by .loc.
+    """
+    from datetime import datetime
+    df = pd.DataFrame(columns=['1'],
+                      index=pd.date_range('2016-10-01T00:00:00',
+                                          '2016-10-01T23:59:59'))
+    df_sliced_dt = df.loc[datetime(2016, 10, 1):]
+    tm.assert_frame_equal(df, df_sliced_dt)
+
+    df = pd.DataFrame(columns=['1'],
```
Comment (TomAugspurger) [inline pandas/tests/indexing/test_loc.py]: This is the same `df` right? Shouldn't need to re-create it (unless I'm missing something).

File: pandas/tests/indexing/test_loc.py [hunk 2]
```diff
@@ -628,3 +628,22 @@ def test_loc_empty_list_indexer_is_ok(self):
         tm.assert_frame_equal(df.loc[[]], df.iloc[:0, :],
                               check_index_type=True,
                               check_column_type=True)
+
+
+def test_loc_datetime_length_one():
+    """
```
Comment (TomAugspurger) [inline pandas/tests/indexing/test_loc.py]: You can replace this docstring with the github issue number

**Review by TomAugspurger — COMMENTED** [review]
I think since you aren't using any of the advanced pytest features like fixtures, it'd be better to include this test with other similar ones on the class.

File: pandas/tests/indexing/test_loc.py [hunk 3]
```diff
@@ -628,3 +628,22 @@ def test_loc_empty_list_indexer_is_ok(self):
...
+    Test if DataFrame with length one is properly sliced
+    by .loc.
+    """
+    from datetime import datetime
```
Comment (jreback) [inline pandas/tests/indexing/test_loc.py]: datetime should already be imported (if not add at the very top)

File: pandas/tests/indexing/test_loc.py [hunk 4]
```diff
@@ -628,3 +628,22 @@ def test_loc_empty_list_indexer_is_ok(self):
...
+    df = pd.DataFrame(columns=['1'],
+                      index=pd.date_range('2016-10-01T00:00:00',
+                                          '2016-10-01T23:59:59'))
+    df_sliced_dt = df.loc[datetime(2016, 10, 1):]
```
Comment (jreback) [inline pandas/tests/indexing/test_loc.py]:
```
result = ....
tm.assert_frame_equal(result, df)
```

File: doc/source/whatsnew/v0.20.0.txt [hunk 5]
```diff
@@ -557,6 +557,7 @@ Other Enhancements
 - The ``display.show_dimensions`` option can now also be used to specify
   whether the length of a ``Series`` should be shown in its repr (:issue:`7117`).
 - ``parallel_coordinates()`` has gained a ``sort_labels`` keyword arg that sorts class labels and the colours assigned to them (:issue:`15908`)
+- ``loc`` indexer with datetime strings now works correctly for length 1 DataFrames and is consistent with the behaviour of using a ``datetime.datetime`` object (:issue:`16071')
```
Comment (jreback) [inline doc/source/whatsnew/v0.20.0.txt]: move to Bug / Conversion section

just say: ``Bug in indexing with partial string indexing with a len-1 ``DataFrame`` ``

File: pandas/tests/indexing/test_loc.py [hunk 6]
```diff
@@ -628,3 +628,22 @@ def test_loc_empty_list_indexer_is_ok(self):
         tm.assert_frame_equal(df.loc[[]], df.iloc[:0, :],
                               check_index_type=True,
                               check_column_type=True)
+
+
```
Comment (jreback) [inline pandas/tests/indexing/test_loc.py]: move this test to : ``pandas/tests/indexes/datetimes/test_partial_slcing.py ``

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by cpaulik** [comment]
Thank you for the feedback. I made the requested changes.

**Review by jreback — APPROVED** [review]

**Pull request closed without merging**